### PR TITLE
feat(server): migrate Zalando rule 171 to use the new context object …

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Related to #714

### CODE REVIEW

File: server/src/main/java/de/zalando/zally/rule/zalando/FormatForNumbersRule.kt [hunk 1]
```diff
@@ -2,52 +2,33 @@ package de.zalando.zally.rule.zalando
...
-        }
-        val result = fromObjects + fromParams + fromPathParams
-        return if (result.isNotEmpty()) {
-            val (props, paths) = result.unzip()
-            val properties = props.flatten().toSet().joinToString(", ")
-            Violation(description + properties, paths)
-        } else null
-    }
+    fun checkNumberFormat(context: Context): List<Violation> =
+        allSchemas(context.api)
```
Comment (ddufourboivin-zalando) [inline server/src/main/java/de/zalando/zally/rule/zalando/FormatForNumbersRule.kt]: That would be a good candidate for a method `validateSchemas` (if not already there) in the context.

If not, I would suggest making it an extension method (again, I think is already a file with a collection of extension methods for `OpenAPI` and/or `Swagger`).

#minorSuggestion ;)

**Comment by ddufourboivin-zalando** [comment]
👍

**Comment by maxim-tschumak** [comment]
:+1:

File: server/src/main/java/de/zalando/zally/rule/zalando/FormatForNumbersRule.kt (hunk 1 above)
Comment (maxim-tschumak) [reply server/src/main/java/de/zalando/zally/rule/zalando/FormatForNumbersRule.kt]: Sorry, I've seen this comment after I've merged. I'll incorporate this change in the next PR

**Pull request merged**